lib/index.js

```diff
@@ -153,11 +128,18 @@ async function optimizeBitmap(sourcePath, format,
 
 	if(scale) {
 		let metadata = await image.metadata();
-		image.resize({ width: metadata.width * scale, height: metadata.height * scale });
+		if(metadata.width && metadata.height) {
+			image.resize({
+				width: metadata.width * scale,
+				height: metadata.height * scale
+			});
```

Comment:
YMMV, but destructuring could make this a little less verbose and thus more readable?

```javascript
let { width, height } = await image.metadata();
if(width && height) {
	image.resize({
		width: width * scale,
		height: height * scale
	});
}
```

In fact, this made me realize a potential bug: Should we worry about zero values in that `if` statement? You never know what kind of weird input people have...